src/Helpers/Goodie.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Engelsystem\Helpers;
+
+use Engelsystem\Database\Database;
+use Engelsystem\Models\User\User;
+use Illuminate\Database\Query\Expression;
+use Illuminate\Database\Query\Grammars\MySqlGrammar;
+
+class Goodie
+{
+    /**
+     * Generates the query to sum night shifts
+     *
+     * Shifts and shift entries must be available via join
+     */
+    public static function nightShiftsSumQuery(): Expression
+    {
+        $nightShifts = config('night_shifts');
+
+        /** @var Database $db */
+        $db = app(Database::class);
+        $connection = $db->getConnection();
+
+        if (!$connection->getQueryGrammar() instanceof MySqlGrammar) {
+            return $connection->raw('0');
+        }
+
+        // @codeCoverageIgnoreStart
+        // as sqlite does not support TIMESTAMPDIFF
+
+        if (!$nightShifts['enabled']) {
+            return $connection->raw(
+                /** @lang MySQL */
+                'COALESCE(SUM(TIMESTAMPDIFF(MINUTE, shifts.start, shifts.end) * 60), 0)'
+            );
+        }
+
+        /* @see \Engelsystem\Models\Shifts\Shift::isNightShift to keep them in sync */
+        $query =
+            /** @lang MySQL */
+            '
+                COALESCE(SUM(
+                    /* Shift length */
+                    TIMESTAMPDIFF(MINUTE, shifts.start, shifts.end) * 60
+
+                    /* Handle freeloading */
+                    * (
+                        1
+                        - (%3$d + 1)
+                        * (
+                            CASE WHEN `shift_entries`.`freeloaded_by` IS NULL
+                            THEN 0
+                            ELSE 1
+                            END
+                        )
+                    )
+
+                    /** Is night shift */
```

Comment:
this only doubles if it's a night shift, it doesn't multiply with the night shift multiplier